File: Aula02/depois/Organico.Library/PedidosCosmosClient.cs
using Microsoft.Azure.Cosmos;
using Organico.Library.Model;

namespace Organico.Library
{
    public class PedidosCosmosClient : BaseCosmosClient
    {
        protected override string ContainerId { get; set; } = "Pedidos";

        private static PedidosCosmosClient? instance;

        private PedidosCosmosClient()
        {
        }

        public static PedidosCosmosClient Instance()
        {
            if (instance == null)
            {
                instance = new PedidosCosmosClient();
                instance.InitializeAsync();
            }
            return instance;
        }

        public async Task<List<Order>> GetList()
        {
            var result = new List<Order>();

            using (FeedIterator<Order?> feedIterator = _container.GetItemQueryIterator<Order?>(
                "select * from Items i"))
            {
                while (feedIterator.HasMoreResults)
                {
                    FeedResponse<Order?> response = await feedIterator.ReadNextAsync();
                    foreach (var responseItem in response)
                    {
                        result.Add(responseItem);
                    }
                }
            }

            return result;
        }

        public async Task Post(Order order)
        {
            var itemResponse = await _container.UpsertItemAsync(order);
        }
    }
}

File: Aula04/depois/Organico.JS/pedidosCosmosClient.js
const { CosmosClient } = require("@azure/cosmos");

class PedidosCosmosClient {
  constructor() {
    this._client = new CosmosClient({
      endpoint: process.env.CosmosDB_URI,
      key: process.env.CosmosDB_KEY
    });

    this._databaseId = 'organico';
  }

  async initializeAsync() {
    this._container = await this._client.database(this._databaseId).container('Pedidos');
  }

  async get(id) {
    const result = [];

    const querySpec = {
      query: `SELECT * FROM c WHERE c.id = '${id}'`
    };

    const { resources: pedidos } = await this._container.items.query(querySpec).fetchNext();
    if (pedidos.length == 0) {
        return null;
    }
    return pedidos[0]
  }

  async post(order) {
    const itemResponse = await this._container.items.upsert(order);
  }
}

module.exports = PedidosCosmosClient;
